fission-web-server/library/Fission/Web/Server/IPFS/Cluster/Error.hs

```diff
@@ -0,0 +1,29 @@
+module Fission.Web.Server.IPFS.Cluster.Error
+  ( module Fission.Web.Server.IPFS.Cluster.Error.Types
+  , parseClientError
+  ) where
+
+import qualified Fission.Internal.UTF8                       as UTF8
+import           Fission.Prelude
+import           Servant.Client
+
+import           Fission.Web.Server.IPFS.Cluster.Error.Types
+
+-- | Parse and Log the Servant Client Error returned from the IPFS Daemon
+parseClientError :: MonadLogger m => ClientError -> m Error
+parseClientError err = do
+  logError <| displayShow err
+  return <| case err of
+    FailureResponse _ response ->
+      response
+        |> responseBody
+        |> decode
+        |> \case
+          Just ErrorBody {message} ->
+            ClusterDaemonErr <| UTF8.textShow message
+
+          _ ->
+            UnexpectedOutput <| UTF8.textShow err
```

Comment:
Two things:

1. We're moving away from `<|`
2. `ClientError` has an (orphan) `Display` instance now

```haskell
parseClientError err = do
  logError $ displayShow err
  case err of
    FailureResponse _ response ->
      case decode $ responseBody response of
          Just ErrorBody {message} ->
            return . ClusterDaemonErr $ textDisplay message

          _ ->
            return . UnexpectedOutput $ textDisplay err
    unknownClientError ->
      UnknownPinErr $ textDisplay unknownClientError
```

Reply:
ah yup you're right. this code was (heavily :stuck_out_tongue:) inspired by code from `ipfs-haskell` which is still using `<|`